doc/source/user/basics.rec.rst

```diff
@@ -576,7 +576,7 @@ Record Arrays
 
 As an optional convenience numpy provides an ndarray subclass,
 :class:`numpy.recarray`, and associated helper functions in the
-:mod:`numpy.lib.recfunctions` submodule (aliased as ``numpy.rec``), that allows
+:mod:`numpy.lib.recfunctions` submodule, that allows
```

Comment:
Hmm, I wonder which part is in error is here.  The `numpy.rec` (`numpy.core.records.py`) has the `record` class that allow access to structured types by attribute, while `numpy.lib.recfunctions` looks like a collection of utility functions.

Reply:
`numpy.lib.recfunctions` does have helper functions for dealing with structured arrays, `numpy.rec` is an alias for `numpy.core.records`. I chose to remove the incorrect statement but both modules could be called out if that's preferred, something like:

```rst
... :class:`numpy.recarray`, and associated helper functions in the
:mod:`numpy.lib.recfunctions` and :mod:`numpy.rec` modules, that allows...

Reply:
I think both should be covered and their differences noted. The `record.py` module is actually imported at the top level, e.g, `numpy.rec` is available, while `recfunctions` needs `import numpy.lib.recfunctions`. There was an inconclusive debate long ago as to whether the `recfunctions` should be moved up a level. I don't know how much either is used or if we should encourage that. @shoyer Does xarray cover some of the same functionality?

Reply:
> I think both should be covered and their differences noted. 

Thanks for the feedback! I gave this a shot in 3599af5.

Reply:
Structured arrays are fine for basic use-cases, but I would encourage users to consider using pandas or xarray instead. Pandas and xarray have far more functionality, are better maintained and use a more efficient implementation under the hood (struct of arrays vs arrays of struct).

I would suggest adding references to pandas and xarray into our docs here, to help point users in the right direction.

Reply:
A bit off topic, and not disagreeing with the recommendation to reference pandas and xarray. However there are many use cases for "arrays of struct". They are widely used in industries that must parse or stream binary data (think transmissions from weather stations) since they are easier to work with and more performant than ctypes Structures. These days CFFI provides a convenient alternative, but NumPy is still the easiest tool to reach for.

Reply:
References to pandas and xarray don't really belong in a section specific to `recarray`, since they're replacements for more general structured arrays. I'd be inclined to leave that change for a follow-up PR.